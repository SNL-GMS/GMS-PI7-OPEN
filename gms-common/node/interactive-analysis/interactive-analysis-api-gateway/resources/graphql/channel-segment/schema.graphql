enum TimeSeriesType {
    WAVEFORM
    FK_SPECTRA
    DETECTION_FEATURE_MAP
}

# Represents a channel time segment comprised of one or more waveforms for a (derived or raw).
type ChannelSegment {

    # Unique identifier for the channel segment
    id: String!

    # Name of ChannelSegment
    name: String!

    # Type of timeseries
    timeseriesType: TimeSeriesType!

    # The type enumeration of the channel segment
    type: ChannelSegmentType

    # Unique identifier for the channel id
    channelId: String!

    # The start date/time (seconds since epoch) of the channel segment
    startTime: Float!

    # The end date/time ((seconds since epoch) of the channel segment
    endTime: Float!

    # The channel associated with the segment
    channel: ProcessingChannel

    # The collection of Waveforms associated with the channel segment
    timeseries: [Timeseries]

    # Provenance creation information for the channel segment
    creationInfo: CreationInfo!
}

# Input version of the time window parameters provided as input to the FK mutation API
input WindowParametersInput {

    # The type of window (e.g. Hanning)
    windowType: String!

    # The lead period in seconds (prior to the detection onset time)
    leadSeconds: Float!

    # The length of the window in seconds (including the lead and lag around the detection onset time)
    lengthSeconds: Float!

    #  The number of FK Spectra per second to fetch
    stepSize: Float!
}

input ContributingChannelsConfigurationInput {
    id: String!
    enabled: Boolean!
    name: String
}

# Holds the configuration used to calculate an Fk
  
input FkConfigurationInput {
    maximumSlowness: Float!
    mediumVelocity: Float!
    numberOfPoints: Float!
    normalizeWaveforms: Boolean!
    useChannelVerticalOffset: Boolean!
    contributingChannelsConfiguration: [ContributingChannelsConfigurationInput]!
    leadFkSpectrumSeconds: Float!
}
# Input parameters used to create a new FK data set
input FkInput {

    # The ID of the station the FK will be created for
    stationId: String!

    # Signal detection that Fk is created from
    signalDetectionId: String!

    # Signal detection hypothesis
    signalDetectionHypothesisId: String!

    # Phase of the signal detection
    phase: String!

    # The frequency band to compute the Fk on
    frequencyBand: FrequencyBandInput!

    # The window parameters to compute the Fk with
    windowParams: WindowParametersInput!

    # Configuration settings for the continous fk
    configuration: FkConfigurationInput
}

# Input to modify leadSeconds and return SD with leadFkSpectrum and PowerSpectrum populated
input FkLeadInput {
     # Signal detection that Fk is created from
    signalDetectionId: String!

    # Window Lead Seconds and Lead Fk Spectrum Select Seconds (movie)
    windowLeadSeconds: Float! # If lead is changed thru the widget
    leadFkSpectrumSeconds: Float!
}

# Input parameters that indicate which Fk (ChannelSegment TimeSeries)
# to set the reviewed flag
input MarkFksReviewedInput {
    # Signal detection that Fk is created from
    signalDetectionIds: [String]!

    # Boolean flag to set if reviewed by the Analyst
    reviewed: Boolean!
}

extend type Query {
    # Compute the Fk Thumbnails for UI Display. Should be called from UI after a
    # new Fk is called
    computeFkFrequencyThumbnails(fkInput: FkInput!): FkFrequencyThumbnailBySDId!
}

# Currently, no queries are provided. Fk data are accessed as members of
# AzslownessFeatureMeasurements on SignalDetectionHypotheses

# The following signal FK mutations are supported.
extend type Mutation {
    setFkWindowLead(leadInput: FkLeadInput!): AssociationChange

    # Compute the new FKs and set for each signal detection id
    computeFks(fkInput: [FkInput]!): AssociationChange

    # update reviewed flag 
    markFksReviewed(markFksReviewedInput: MarkFksReviewedInput!): AssociationChange

    # Calculate new Beam (replaces ArrivalTime FM Beam)
    computeBeam(fkInput: FkInput!): AssociationChange
}
