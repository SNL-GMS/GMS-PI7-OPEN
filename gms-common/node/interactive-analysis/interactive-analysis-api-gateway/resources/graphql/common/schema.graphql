# Common GraphQL schema definitions shared across the components of the
# API Gateway (e.g. waveform, workflow)

# Units used in DoubleValue part of feature prediction
enum Units {
    DEGREES
    SECONDS
    SECONDS_PER_DEGREE
    UNITLESS
}

# Represents Double fields for a double value representation.
type DoubleValue {
    value: Float # Optional Guidance is value can be a NaN i.e. FP predicted value
    standardDeviation: Float!
    units: Units!
}

# Represents Double fields for a double value representation input from UI.
input DoubleValueInput {
    value: Float!
    standardDeviation: Float!
    units: Units!
}

# Phase types
enum PhaseType {
    P
    S
    P3KPbc
    P4KPdf_B
    P7KPbc
    P7KPdf_D
    PKiKP
    PKKSab
    PKP2bc
    PKP3df_B
    PKSab
    PP_1
    pPKPbc
    PS
    Rg
    SKiKP
    SKKSac
    SKPdf
    SKSdf
    sPdiff
    SS
    sSKSdf
    Lg
    P3KPbc_B
    P5KPbc
    P7KPbc_B
    Pb
    PKKP
    PKKSbc
    PKP2df
    PKPab
    PKSbc
    PP_B
    pPKPdf
    PS_1
    SKKP
    SKKSac_B
    SKS
    SKSSKS
    sPKiKP
    SS_1
    SSS
    nNL
    P3KPdf
    P5KPbc_B
    P7KPbc_C
    PcP
    PKKPab
    PKKSdf
    PKP3
    PKPbc
    PKSdf
    pPdiff
    PPP
    pSdiff
    Sb
    SKKPab
    SKKSdf
    SKS2
    Sn
    sPKP
    SS_B
    SSS_B
    NP
    P3KPdf_B
    P5KPdf
    P7KPdf
    PcS
    PKKPbc
    PKP
    PKP3ab
    PKPdf
    Pn
    pPKiKP
    PPP_B
    pSKS
    ScP
    SKKPbc
    SKP
    SKS2ac
    SnSn
    sPKPab
    sSdiff
    NP_1
    P4KPbc
    P5KPdf_B
    P7KPdf_B
    Pdiff
    PKKPdf
    PKP2
    PKP3bc
    PKPPKP
    PnPn
    pPKP
    PPS
    pSKSac
    ScS
    SKKPdf
    SKPab
    SKS2df
    SP
    sPKPbc
    sSKS
    P4KPdf
    P5KPdf_C
    P7KPdf_C
    Pg
    PKKS
    PKP2ab
    PKP3df
    PKS
    PP
    pPKPab
    PPS_B
    pSKSdf
    Sdiff
    SKKS
    SKPbc
    SKSac
    SP_1
    sPKPdf
    sSKSac
    Sx
    tx
    N
    Px
    PKhKP
    LR
    pP
    UNKNOWN
}

# A date/time Type
scalar Date

# Represents a time range
# type TimeRange {
#     startTime: Float!
#     endTime: Float!
# }

# Input verion of a time range
input TimeRange {
    startTime: Float!
    endTime: Float!
}

# Represents a location specified using latitude (degrees), longitude (degrees),
# and altitude (kilometers).
type Location {

    # Latitude in degrees
    latDegrees: Float!
    
    # Longitude in degrees
    lonDegrees: Float!

    # Elevation in kilometers
    elevationKm: Float!
}

input LocationInput {

    # Latitude in degrees
    latDegrees: Float!
    
    # Longitude in degrees
    lonDegrees: Float!

    # Elevation in kilometers
    elevationKm: Float!
}

# Distance value's units degrees or kilometers
enum DistanceUnits {
    degrees
    km
}

# Enumerated list of source types used to compute distances to
enum DistanceSourceType {
    Event
    UserDefined
}

# Represents a distance measurement relative to a specified source location
type DistanceToSource {

    # The distance in kilometers or degrees to the source
    distance: Float!

    # The units distance was calculated
    distanceUnits: DistanceUnits

    # The source location
    sourceLocation: Location!

    # The type of the source the distance is measured to (e.g. and event)
    sourceType: DistanceSourceType!

    # Optional: the unique ID of the source object
    sourceId: String!

    # Which station distance to the source
    stationId: String!
}

input DistanceToSourceInput {
    # The source location
    sourceLocation: LocationInput

    # The type of the source the distance is measured to (e.g. and event)
    sourceType: DistanceSourceType

    # The Distance Units to use
    distanceUnits: DistanceUnits

    # Optional: the unique ID of the source object
    sourceId: String
}

# Represents a frequency range
type FrequencyBand {
    # Minimum frequency in Herz
    minFrequencyHz: Float!

    # Maximum frequency in Herz
    maxFrequencyHz: Float!
}

# Represents a frequency range input to a mutation
input FrequencyBandInput {
    # Minimum frequency in Herz
    minFrequencyHz: Float!

    # Maximum frequency in Herz
    maxFrequencyHz: Float!
}

# Enumeration representing the different types of stations in the monitoring network
enum StationType {

    Seismic3Component,
    Seismic1Component,
    SeismicArray,
    Hydroacoustic,
    HydroacousticArray,
    Infrasound,
    InfrasoundArray,
    Weather,
    UNKNOWN
}

# Type enumeration for channel segments
enum ChannelSegmentType {
    ACQUIRED
    RAW
    DETECTION_BEAM
    FK_BEAM
    FILTER
    FK_SPECTRA
}

# Provenance reference details for information added to the System
type InformationSource {

    # Unique ID for the source
    id: String!

    # Source organization
    organizationName: String!

    # The time associated with the information source
    referenceTime: Float!
}

# Enumeration of configuration status changes reflected in the configuration
enum ConfigurationInfoStatus {

    # Initial configuration
    Initial

    # The system's installed location for the item associated with the configuration was corrected
    CorrectedSystemLocation

    # The physical location of the ite associated with the configuration was changed
    MovedPhysicalLocation

    # The item associated with the configuration was decommissioned
    Decommissioned
}

# Provenance information associated with a configuration item in the system
type ConfigurationInfo {

    # The unique ID of the user who created the Configuration
    creatorId: String!

    # The time the configuration took effect in the real world (e.g. station came online)
    effectiveTime: Float!

    # The time the configuration was installed (and thus took effect) in the System
    installationTime: Float!

    # The status change reflected in the configuration values
    status: ConfigurationInfoStatus!

    # Comment associated with the configuration information
    comment: String
}

# Enumerated type of the actor (e.g. analyst, system) associated with a CreationInfo object
enum CreatorType {
    Analyst
    System
}

# Provenance information about the results of data processing completed by the System
# and actions completed by users of the system
type CreationInfo {
    # The unique ID of the creation info object
    id: String!

    # The time (seconds since Unix epoch) the associated processing result was created
    creationTime: Float!

    # The unique ID of the user who created the associated processing result
    creatorId: String!

    # The enumerated type of the creator (e.g. analyst, system)
    creatorType: CreatorType!

     # The name of the creation
    creatorName: String!

    # TODO
    # The processing step associated with the processing result
    # processingStep: ProcessingStep!

    # TODO
    # The software component associated with the processing result
    # softwareComponent: SoftwareComponent!
}

input CreationInfoInput {
    # Identifier for the creation info
    creatorName: String!
    
    # The time (seconds since Unix epoch) the associated processing result was created
    creationTime: Float!

    # The software information associated with the processing result
    softwareInfo: SoftwareInfoInput!
}

type SoftwareInfo {
    # Identifier for the creation info
    name: String!

    # The software version identifier
    version: String!
}

input SoftwareInfoInput {
    # Identifier for the creation info
    name: String!

    # The software version identifier
    version: String!
}

input ProcessingContextInput {
    analystActionReference: AnalystActionReferenceInput!
    processingStepReference: ProcessingStepReferenceInput!
    storageVisibility: String!
}

input AnalystActionReferenceInput {
    processingStageIntervalId: String!
    processingActivityIntervalId: String!
    analystId: String!
}

input ProcessingStepReferenceInput {
    processingStageIntervalId: String!
    processingSequenceIntervalId: String!
    processingStepId: String!
}

# Represents an alias for a station, channel, site, etc.
type Alias {

    # The alias name
    name: String!

    # Provenance information about the alias
    configurationInfo: ConfigurationInfo
}

# Represents calibration information associated with a waveform
type ProcessingCalibration {

    # Calibration factor
    factor: Float!

    # Calibration factor error
    factorError: Float!

    # Calibration period
    period: Float!

    # Calibration time shift
    timeShift: Float!
}

# Input version of calibration information associated with a waveform
input ProcessingCalibrationInput {

    # Calibration factor
    factor: Float!

    # Calibration factor error
    factorError: Float!

    # Calibration period
    period: Float!

    # Calibration time shift
    timeShift: Float!
}

# Represents a generic timeseries interface containing the fields common to the
# concrete timeseries data types defined in the system (e.g. Waveform, FkPowerSpectrum, DetectionFeatureMap)
interface Timeseries {
    # The start date/time (seconds since epoch) of the timeseries
    startTime: Float!

    # The sample rate (Hz) of the samples in the timeseries
    sampleRate: Float!

    # The sample count in the timeseries
    sampleCount: Float!
}

input TimeseriesInput {
    # The start date/time (seconds since epoch) of the timeseries
    startTime: String!
    
    # The sample rate (Hz) of the samples in the timeseries
    sampleRate: Float!
    
    # The sample count in the timeseries
    sampleCount: Float!
}

# Represents a time-bounded segment of waveform samples output from a channel (derived or raw).
# Waveforms implement the Timeseries interface
type Waveform implements Timeseries {
    # Unique identifier for the waveform
    # id: String!

    # The start date/time (seconds since epoch) of the waveform
    startTime: Float!

    # The sample rate (Hz) of the samples in the waveform
    sampleRate: Float!

    # The sample count in the waveform
    sampleCount: Float!

    # The time-ordered collection of waveform sample values
    values: [Float]
}

# Input version of a time-bounded segment of waveform samples output from a channel (derived or raw).
# Waveforms implement the Timeseries interface
input WaveformInput {
    
    # Unique identifier for the waveform
    id: String!

    # The start date/time (seconds since epoch) of the waveform
    startTime: Float!

    # The end date/time (seconds since epoch) of the waveform
    endTime: Float!    # The sample rate (Hz) of the samples in the waveform
    sampleRate: Float!

    # The sample count in the waveform
    sampleCount: Float!

    # The time-ordered collection of waveform sample values
    values: [Float]

    # Calibration information for the waveform
    calibration: ProcessingCalibrationInput
}

# The following queries are supported
type Query {
    # Dummy
    queryCount: Int
}

# The following mutations are supported
type Mutation {
    # Dummy
    mutationCount: Int
}

# The following subscriptions are supported
type Subscription {
    # Dummy
    subscriptionCount: Int
}

# Association Change object to return
type AssociationChange {
    events: [Event]
    sds: [SignalDetection]
}

 # Used when we need to return the original event mutated where performed on
 # in addition to the other data changed
type EventAndAssociationChange {
    event: Event!
    associationChange: AssociationChange!
}

#
enum LogLevel {
    INFO
    DEBUG
    WARNING
    ERROR
    DATA
}

type ClientLog {
    logLevel: LogLevel
    message: String!
    time: String
}

input ClientLogInput {
    logLevel: LogLevel!
    message: String!
    time: String
}

extend type Mutation {
    clientLog(clientLogInput: ClientLogInput): ClientLog
}