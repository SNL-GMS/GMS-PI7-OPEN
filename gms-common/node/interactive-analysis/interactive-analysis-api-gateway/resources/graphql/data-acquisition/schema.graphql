
# Represents the transfer gap fields used for gap status.
type FileGap {
    # Station Id
    stationName: String

    # Channel Id
    channelNames: [String]

    # Start time for transfer gap
    startTime: String

    # End time for transfer gap
    endTime: String

    # Duration for transfer gap
    duration: String

    # Location for transfer gap
    location: String

    # Priority for transfer gap
    priority: String
}

# The following queries are supported.
extend type Query {
    # Retrieve transfer file information by time range
    transferredFilesByTimeRange(timeRange: TimeRange!): [FileGap]
}

# Input to Reference Station mutations
input ReferenceStation {
    name: String!
    description: String!
    stationType: StationType!
    comment: String!
    source: InformationSourceInput!
    latitude: Float!
    longitude: Float!
    elevation: Float!
    actualChangeTime: String!
    systemChangeTime: String!
    aliases: [ReferenceAliasInput]!
}

input InformationSourceInput {
    originatingOrganization: String
    informationTime: String
    reference: String
}

input ReferenceAliasInput {
    id: String
    name: String
    status: StatusTypeName
    comment: String
    actualChangeTime: String
    systemChangeTime: String
}

enum StatusTypeName {
    INACTIVE,
    ACTIVE
}

type SaveStationResult{
    id: String!
    result: Boolean!
}

# The following Data Aquisition mutations are supported.
extend type Mutation {
    saveReferenceStation (input: ReferenceStation!): SaveStationResult
}