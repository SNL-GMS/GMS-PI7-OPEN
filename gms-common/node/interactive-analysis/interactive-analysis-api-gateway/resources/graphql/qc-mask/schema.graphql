# Represents a qc mask for a time segment for one or more channels.
type QcMask {

    # Unique identifier of QC mask
    id: String!

    # The id for the processing channel
    channelId: String!

    # QC Mask Version 
    qcMaskVersions: [QcMaskVersion]!

    # Current Mask Version
    currentVersion: QcMaskVersion!
},

# QC mask information
type QcMaskVersion {
    category: String!
    channelSegmentIds: [String]!
    creationInfo: CreationInfo!
    startTime: Float!
    endTime: Float!
    parentQcMasks: [QcMask]!
    rationale: String!
    type: String
    version: Int!
},

extend type Query {
    # Get all QcMasks for each channel that that fall within the start and endtime
    # If no channelIds are provided, Get all masks in the specified time range
    qcMasksByChannelId(timeRange: TimeRange!, channelIds: [String]): [QcMask]
},

# Input to QC Mask mutations
input QcMaskInput {
    timeRange: TimeRange

    category: String

    type: String

    rationale: String
}

# The following QC Mask mutations are supported.
extend type Mutation {
    # Creates new QC Masks
    createQcMask (channelIds: [String]!, input: QcMaskInput!): [QcMask]

    # Update an existing QC Mask using the provided input
    updateQcMask (qcMaskId: String!, input: QcMaskInput!): [QcMask]

    # Reject an existing QC Mask
    rejectQcMask (qcMaskId: String!, rationale: String): [QcMask]  
}

# The following subscription are supported.
extend type Subscription {
    # Subscribe for updates to QC masks on the specified channels in the
    # specified time range
    # If no parameters are provided, subscribe for all mask updates

    # Subscription for QC Masks created
    qcMasksCreated (timeRange: TimeRange, channelIds: [String]): [QcMask]   
}