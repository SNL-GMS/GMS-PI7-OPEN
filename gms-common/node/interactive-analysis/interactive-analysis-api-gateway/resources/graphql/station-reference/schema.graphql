# Represents a group of stations used for monitoring.
type Network {

    # Unique identifier for the network
    id: String!

    # Name of the network
    name: String

    # The monitoring organization the station is affiliated with
    monitoringOrganization: String

    # The geographic region associated with the network
    region: String

    # Provenance information about the network configuration
    configurationInfo: ConfigurationInfo

    # Source of network configuration information
    informationSource: InformationSource

    # Many-to-many association object between the network and associated stations
    # The association object is included to record provenance information
    networkMemberships: [NetworkMembership]
}

# Represents a many-to-may relationship between networks and stations
# with associated provenance information recorded in the associated
# configuration information
type NetworkMembership {

    # Unique identifier
    id: String!

    # Provenance information for this configuration item
    configurationInfo: ConfigurationInfo!

    # Station associated with the network
    station: Station!

    # Network associated with the station
    network: Network!
}

# Represents a many-to-may relationship between stations & sites
# with associated provenance information recorded in the associated
# configuration information
type StationMembership {

    # Unique identifier
    id: String!

    # Provenance information for this configuration item
    configurationInfo: ConfigurationInfo!

    # Site associated with the station
    site: Site!

    # Station associated with the site
    station: Station!
}

# An installation where monitoring sensors are installed.
# Multiple sensors can be installed at the same station.
type Station {

    # Unique identifier for the station
    id: String!

    # Name of the station
    name: String

    # Description of the station
    description: String

    # The type of station (seismic, hydroacoustic, infrasonic)
    stationType: StationType

    # Provenance information about the station configuration
    configurationInfo: ConfigurationInfo

    # Source of station configuration information
    informationSource: InformationSource

    # Location of the station
    location: Location!

    # Aliases for the site
    aliases: [Alias]

    # Many-to-many association object between the station and associated networks
    # The association object is included to record provenance information
    networkMemberships: [NetworkMembership]

    # Many-to-many association object between the station and associated sites
    # The association object is included to record provenance information
    stationMemberships: [StationMembership]

    # Latitude of the station
    latitude: Float

    # Longitude of the station
    longitude: Float

    # Elevation of the station
    longitude: Float
}



# Represents a many-to-may relationship between sites & channels
# with associated provenance information recorded in the associated
# configuration information
type SiteMembership {

    # Unique identifier
    id: String!

    # Provenance information for this configuration item
    configurationInfo: ConfigurationInfo!

    # Site associated with the channel
    site: Site!

    # channel associated with the site
    channel: Channel!
}


# Represents a physical installation (e.g., building, underground vault, borehole)
# containing a collection of Instruments that produce Raw Channel waveform data.
type Site {

    # Unique identifier for the site
    id: String!

    # The name of the site
    name: String

    # Provenance information about the site configuration
    configurationInfo: ConfigurationInfo

    # Source of site configuration information
    informationSource: InformationSource

    # Location of the site
    location: Location!

    # Aliases for the site
    aliases: [Alias]

    # Many-to-many association object between the site and associated stations
    # The association object is included to record provenance information
    stationMemberships: [StationMembership]

    # Many-to-many association object between the site and associated channels
    # The association object is included to record provenance information
    stationMemberships: [StationMembership]
}

# Represents an measurement instrument associated with a raw channel.
type Instrument {

    #  Unique identifier for the instrument
    id: String!

    # The name of the instrument manufacturer
    manufacturer: String

    # The name of the instrument model
    model: String

    # The instrument serial number
    serialNumber: String

    # Corner period of the instrument
    cornerPeriod: Float

    # Provenance information about the instrument configuration
    configurationInfo: ConfigurationInfo

    # Source of instrument configuration information
    informationSource: InformationSource
}

# Represents a source for unprocessed (raw) or processed (derived) time series data
# from a seismic, hydroacoustic, or infrasonic sensor
type Channel {

    # Unique identifier for the channel
    id: String!

    # Name of the channel
    name: String!

    # The type of the channel
    channelType: ChannelType!

    # Location code of the channel
    locationCode: String

    # The vertical angle (degrees) of the channel
    verticalAngle: Float

    # The horizontal angle (degrees) of the channel
    horizontalAngle: Float

    # The sample rate of the channel
    sampleRate: Float

    # The depth of the channel
    depth: Float

    # The instrument associated with the channel
    instrument: Instrument

    # Provenance information about the channel configuration
    configurationInfo: ConfigurationInfo

    # Source of channel configuration information
    informationSource: InformationSource

    # List of aliases for the channel
    aliases: [Alias]

}

# The following channel-related queries are supported.
extend type Query {

    # Retrieve the default channel list for use in automated and interactive analysis
    defaultChannels: [Channel]
}