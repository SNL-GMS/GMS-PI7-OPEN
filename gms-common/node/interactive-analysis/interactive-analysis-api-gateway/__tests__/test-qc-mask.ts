// tslint:disable:max-line-length
import { graphql } from 'graphql';
import { schema } from '../src/ts/schema';
import { delayExecution } from '../src/ts/util/delay-execution';

const delayMs = 500;

// ---- Query test cases ----

// Test case - basic content query for waveforms
// Query for channel AS01/SHZ
it('QC Content query should match snapshot', async () => {
  // language=GraphQL
  const query = `
  query qcmasksbychannel {
    channelsById(ids: ["channel1-1111-1111-1111-111111111110"]) {
      qcMasks(timeRange: {startTime: 1274392801, endTime: 1274396401}) {
        currentVersion {
          type
          category
          startTime
          endTime
        }
      }
    }
  }
  `;

  const rootValue = {};
  // Execution the GraphQL query with a small delay to allow the API gateway to settle async
  // HTTP requests fetching data from the mock backend
  const result = await delayExecution(() => graphql(schema, query, rootValue), delayMs);
  const { data } = result;

  // Compare response to snapshot
  expect(data).toMatchSnapshot();
});

// eb1f82bb-ed5c-4f6b-873c-0b9515aef272

// ---- Mutation test cases ----

// Test case - basic content query for waveforms
// Query for channel AS01/SHZ
it('QC Create mutation should match snapshot', async () => {
  // language=GraphQL
  const query = `
  mutation createMask {
    createQcMask(channelIds: ["channel1-1111-1111-1111-111111111110"], input: {timeRange: {startTime: 1274393921, endTime: 1274393926}, category: "ANALYST_DEFINED", type: "SPIKE", rationale: "testing"}) {
      channelId
      currentVersion {
        startTime
        endTime
        category
        type
        rationale
        version
        channelSegmentIds
      }
      qcMaskVersions {
        startTime
        endTime
        category
        type
        rationale
        version
        channelSegmentIds
      }
    }
  }
  `;

  const rootValue = {};
  // Execution the GraphQL query with a small delay to allow the API gateway to settle async
  // HTTP requests fetching data from the mock backend
  const result = await delayExecution(() => graphql(schema, query, rootValue), delayMs);
  const { data } = result;

  // Compare response to snapshot
  expect(data).toMatchSnapshot();
});

// Test case - basic content query for waveforms
// Query for channel AS01/SHZ
it('QC Update mutation should match snapshot', async () => {
  // language=GraphQL
  // mask Id corresponds a mask on channel AS01/SHZ
  const query = `
  mutation updateMask {
    updateQcMask(qcMaskId: "qcmask11-1111-1111-1111-111111111110", input: {timeRange: {startTime: 1274393921, endTime: 1274393936}, category: "ANALYST_DEFINED", type: "SPIKE", rationale: "Updating Test"}) {
      id
      channelId
      currentVersion {
        startTime
        endTime
        category
        type
        rationale
        version
        channelSegmentIds
      }
      qcMaskVersions {
        startTime
        endTime
        category
        type
        rationale
        version
        channelSegmentIds
      }
    }
  }
  `;

  const rootValue = {};
  // Execution the GraphQL query with a small delay to allow the API gateway to settle async
  // HTTP requests fetching data from the mock backend
  const result = await delayExecution(() => graphql(schema, query, rootValue), delayMs);
  const { data } = result;

  // Compare response to snapshot
  expect(data).toMatchSnapshot();
});

// Test case - basic content query for waveforms
// Query for channel AS01/SHZ
it('QC Create reject should match snapshot', async () => {
  // language=GraphQL
  // mask Id corresponds a mask on channel AS01/SHZ
  const query = `
  mutation rejectMask {
    rejectQcMask(qcMaskId: "qcmask11-1111-1111-1111-111111111110", rationale: "This was a crappy mask anyway") {
      id
      channelId
      currentVersion {
        startTime
        endTime
        category
        type
        rationale
        version
        channelSegmentIds
      }
      qcMaskVersions {
        startTime
        endTime
        category
        type
        rationale
        version
        channelSegmentIds
      }
    }
  }
  `;

  const rootValue = {};
  // Execution the GraphQL query with a small delay to allow the API gateway to settle async
  // HTTP requests fetching data from the mock backend
  const result = await delayExecution(() => graphql(schema, query, rootValue), delayMs);
  const { data } = result;

  // Compare response to snapshot
  expect(data).toMatchSnapshot();
});
