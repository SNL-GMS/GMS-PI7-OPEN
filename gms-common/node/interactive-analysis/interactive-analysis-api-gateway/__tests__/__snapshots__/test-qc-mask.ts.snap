// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QC Content query should match snapshot 1`] = `
Object {
  "channelsById": Array [
    Object {
      "qcMasks": Array [
        Object {
          "currentVersion": Object {
            "category": "WAVEFORM_QUALITY",
            "endTime": 1274395500,
            "startTime": 1274394600,
            "type": "LONG_GAP",
          },
        },
        Object {
          "currentVersion": Object {
            "category": "WAVEFORM_QUALITY",
            "endTime": 1274397000,
            "startTime": 1274396400,
            "type": "LONG_GAP",
          },
        },
      ],
    },
  ],
}
`;

exports[`QC Create mutation should match snapshot 1`] = `
Object {
  "createQcMask": Array [
    Object {
      "channelId": "channel1-1111-1111-1111-111111111110",
      "currentVersion": Object {
        "category": "ANALYST_DEFINED",
        "channelSegmentIds": Array [
          "channel1-1111-1111-1111-111111111110",
        ],
        "endTime": 1274393926,
        "rationale": "testing",
        "startTime": 1274393921,
        "type": "SPIKE",
        "version": 0,
      },
      "qcMaskVersions": Array [
        Object {
          "category": "ANALYST_DEFINED",
          "channelSegmentIds": Array [
            "channel1-1111-1111-1111-111111111110",
          ],
          "endTime": 1274393926,
          "rationale": "testing",
          "startTime": 1274393921,
          "type": "SPIKE",
          "version": 0,
        },
      ],
    },
  ],
}
`;

exports[`QC Create reject should match snapshot 1`] = `
Object {
  "rejectQcMask": Array [
    Object {
      "channelId": "channel1-1111-1111-1111-111111111110",
      "currentVersion": Object {
        "category": "REJECTED",
        "channelSegmentIds": Array [
          "f09572e5-979f-3e07-a181-c1dd75c356b9",
        ],
        "endTime": 1274393936,
        "rationale": "This was a crappy mask anyway",
        "startTime": 1274393921,
        "type": "SPIKE",
        "version": 2,
      },
      "id": "qcmask11-1111-1111-1111-111111111110",
      "qcMaskVersions": Array [
        Object {
          "category": "WAVEFORM_QUALITY",
          "channelSegmentIds": Array [
            "f09572e5-979f-3e07-a181-c1dd75c356b9",
          ],
          "endTime": 1274395500,
          "rationale": "",
          "startTime": 1274394600,
          "type": "LONG_GAP",
          "version": 0,
        },
        Object {
          "category": "ANALYST_DEFINED",
          "channelSegmentIds": Array [
            "f09572e5-979f-3e07-a181-c1dd75c356b9",
          ],
          "endTime": 1274393936,
          "rationale": "Updating Test",
          "startTime": 1274393921,
          "type": "SPIKE",
          "version": 1,
        },
        Object {
          "category": "REJECTED",
          "channelSegmentIds": Array [
            "f09572e5-979f-3e07-a181-c1dd75c356b9",
          ],
          "endTime": 1274393936,
          "rationale": "This was a crappy mask anyway",
          "startTime": 1274393921,
          "type": "SPIKE",
          "version": 2,
        },
      ],
    },
  ],
}
`;

exports[`QC Update mutation should match snapshot 1`] = `
Object {
  "updateQcMask": Array [
    Object {
      "channelId": "channel1-1111-1111-1111-111111111110",
      "currentVersion": Object {
        "category": "ANALYST_DEFINED",
        "channelSegmentIds": Array [
          "f09572e5-979f-3e07-a181-c1dd75c356b9",
        ],
        "endTime": 1274393936,
        "rationale": "Updating Test",
        "startTime": 1274393921,
        "type": "SPIKE",
        "version": 1,
      },
      "id": "qcmask11-1111-1111-1111-111111111110",
      "qcMaskVersions": Array [
        Object {
          "category": "WAVEFORM_QUALITY",
          "channelSegmentIds": Array [
            "f09572e5-979f-3e07-a181-c1dd75c356b9",
          ],
          "endTime": 1274395500,
          "rationale": "",
          "startTime": 1274394600,
          "type": "LONG_GAP",
          "version": 0,
        },
        Object {
          "category": "ANALYST_DEFINED",
          "channelSegmentIds": Array [
            "f09572e5-979f-3e07-a181-c1dd75c356b9",
          ],
          "endTime": 1274393936,
          "rationale": "Updating Test",
          "startTime": 1274393921,
          "type": "SPIKE",
          "version": 1,
        },
      ],
    },
  ],
}
`;
